src/pages/Purchase/Purchase.jsx

```diff
@@ -1,7 +1,167 @@
-import React from "react";
+import PageLoading from "components/Loading/PageLoading";
+import React, { useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { Link, Navigate, NavLink, useParams } from "react-router-dom";
+import SeatItem from "./SeatItem";
+
+import { resetTickets, ticketsByShowtime, bookSelectedTickets } from "redux/slices/ticketsSlice";
+import { Button, notification } from "antd";
+import PopupModal from "components/Modal/PopupModal";
+import useModalHook from "components/Modal/useModalHook";
 
 const Purchase = () => {
-  return <div>Purchase</div>;
+  const dispatch = useDispatch();
+  const { showtimeId } = useParams();
+
+  const { visible, showModal, closeModal } = useModalHook();
+
+  const currencyFormat = new Intl.NumberFormat("vn-VN");
+
+  const { ticketsData, selectedSeats, isLoading, bookedSuccess, error } = useSelector((state) => {
+    return state.tickets;
+  });
+
+  const { currentUser } = useSelector((state) => {
+    return state.auth;
+  });
+
+  let total = selectedSeats?.reduce((total, seat, index) => {
+    return total + seat.giaVe;
+  }, 0);
+
+  const handleSubmit = () => {
+    if (!Object.keys(currentUser).length) {
+      showModal();
+    } else {
+      dispatch(bookSelectedTickets(showtimeId));
+    }
+  };
+
+  useEffect(() => {
+    dispatch(ticketsByShowtime(showtimeId));
+  }, []);
+
+  // useEffect(() => {
+  //   dispatch(resetTickets());
+  // }, [showtimeId]);
+
+  useEffect(() => {
+    if (bookedSuccess) {
+      notification["success"]({ message: "Đặt vé thành công", description: "Sang trang chi tiết vé đã đặt" });
+    }
+  }, [bookedSuccess]);
+
+  if (isLoading) {
+    return <PageLoading />;
```

Comment:
Should use global loading property implementation to get consistent structure (if have time).